### PR TITLE
Add support for idea plugins auto-install

### PR DESCRIPTION
**Pull request by waterfoul**
## Proposed Changes

1. Adds additional logic to python and additional yaml to support creating a file which triggers intellij based IDEs to install the plugins

### Author/Contributor
- [x] Add entry to the [CHANGELOG](https://github.com/oxsecurity/megalinter/blob/main/CHANGELOG.md) listing the change and linking to the corresponding issue (if appropriate)
- [x] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [x] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Review by echoix — APPROVED** [review]

File: docs/reporters/ConfigReporter.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ It contains:
 
 - All default MegaLinter configuration files used during the linting
 - Updated `.vscode/extensions.json` containing VsCode extensions corresponding to the linters used during MegaLinter run
+- Updated `.idea/externalDependencies.xml` containing VsCode extensions corresponding to the linters used during MegaLinter run
```
Comment (echoix) [inline docs/reporters/ConfigReporter.md:13]: I think it won't be VsCode extensions here

**Review by echoix — CHANGES_REQUESTED** [review]

File: docs/reporters/ConfigReporter.md (hunk 1 above)
Comment (waterfoul) [reply docs/reporters/ConfigReporter.md:13]: Good catch, sorry for the typo

File: docs/reporters/ConfigReporter.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ It contains:
 
 - All default MegaLinter configuration files used during the linting
 - Updated `.vscode/extensions.json` containing VsCode extensions corresponding to the linters used during MegaLinter run
+- Updated `.idea/externalDependencies.xml` containing Intellij extensions corresponding to the linters used during MegaLinter run
```
Comment (echoix) [inline docs/reporters/ConfigReporter.md:13]:
```suggestion
- Updated `.idea/externalDependencies.xml` containing IntelliJ extensions corresponding to the linters used during MegaLinter run
```

**Review by nvuillam — CHANGES_REQUESTED** [review]
Seems great, thanks for the PR :)
Please can you add cspell exceptions into .cspell json instead of inline comments ?
MegaLinter provides a reasy to ise updated cspell file in artifacts 
Thanks :)

**Comment by waterfoul** [comment]
> Seems great, thanks for the PR :) Please can you add cspell exceptions into .cspell json instead of inline comments ? MegaLinter provides a reasy to ise updated cspell file in artifacts Thanks :)

Which .cspell.json file should I update? There are a few and none of them are at the root

**Comment by Kurt-von-Laven** [comment]
[.github/linters/.cspell.json](https://github.com/oxsecurity/megalinter/blob/main/.github/linters/.cspell.json)

**Review by echoix — APPROVED** [review]

**Review by nvuillam — APPROVED** [review]

**Pull request merged**